setup.py

```diff
@@ -54,11 +54,11 @@
         "mike",
         "pymdown-extensions",
         # linting
-        "black",
-        "pre-commit",
-        "pylint",
-        "flake8-unused-arguments",
-        "pyright",
+        "black>=23.7.0,<24.0.0",
```

Comment:
for black, the source of truth is the one in the precommit config. To keep the formatting consistent between vscode and the precommits, we may ensure that the versions are in sync (out of the scope of this PR).